src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -421,68 +496,82 @@ namespace OpenMS
   }
 
 
-  pair<IdentificationData::ScoreTypeRef, bool>
+  IdentificationData::ScoreTypeRef
   IdentificationData::findScoreType(const String& score_name) const
   {
     for (ScoreTypeRef it = score_types_.begin(); it != score_types_.end(); ++it)
     {
       if (it->cv_term.getName() == score_name)
       {
-        return make_pair(it, true);
+        return it;
       }
     }
-    return make_pair(score_types_.end(), false);
+    return score_types_.end();
   }
 
 
-  vector<IdentificationData::QueryMatchRef>
-  IdentificationData::getBestMatchPerQuery(ScoreTypeRef score_ref) const
+  vector<IdentificationData::ObservationMatchRef>
+  IdentificationData::getBestMatchPerObservation(ScoreTypeRef score_ref,
+                                                 bool require_score) const
   {
-    vector<QueryMatchRef> results;
-    bool higher_better = score_ref->higher_better;
+    vector<ObservationMatchRef> results;
     pair<double, bool> best_score = make_pair(0.0, false);
-    QueryMatchRef best_ref = query_matches_.end();
-    for (QueryMatchRef ref = query_matches_.begin();
-         ref != query_matches_.end(); ++ref)
+    ObservationMatchRef best_ref = observation_matches_.end();
+    Size n_matches = 1; // number of matches for current observation
+    // matches for same observation appear consecutively, so just iterate:
```

Comment:
How is this ensured and tested so we don't regress on that property by accident?

Reply:
It's a consequence of how the matches are indexed in their multi-index container (ordered by observation first). I've extended the comment there.